api/utils/grpc/interceptors/errors.go

```diff
@@ -50,22 +50,24 @@ type grpcClientStreamWrapper struct {
 
 // SendMsg wraps around ClientStream.SendMsg
 func (s *grpcClientStreamWrapper) SendMsg(m interface{}) error {
-	if err := s.ClientStream.SendMsg(m); err != nil {
-		return &RemoteError{Err: trace.Unwrap(trail.FromGRPC(s.ClientStream.SendMsg(m)))}
-	}
-	return nil
+	return wrapStreamErr(s.ClientStream.SendMsg(m))
 }
 
 // RecvMsg wraps around ClientStream.RecvMsg
 func (s *grpcClientStreamWrapper) RecvMsg(m interface{}) error {
-	switch err := s.ClientStream.RecvMsg(m); {
+	return wrapStreamErr(s.ClientStream.RecvMsg(m))
+}
+
+func wrapStreamErr(err error) error {
+	switch {
+	case err == nil:
+		return nil
 	case errors.Is(err, io.EOF):
```

Comment:
Technically we only need to avoid wrapping if `err == io.EOF`, not if `errors.Is(io.EOF)`, right?

Reply:
It's... difficult to say. If we assume there are no layers between us and the _real_ gRPC stream, then yes. If that changes, then depends. I had an `==` comparison in the original PR and got asked to change to errors.Is for future proofing.